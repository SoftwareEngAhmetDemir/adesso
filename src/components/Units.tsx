import { Fragment, useLayoutEffect } from 'react';
import Header from './Header';
import AgeFilter from './AgeFilter';
import CostFilter from './CostFilter';
import FiteredTable from './FiteredTable';
import { action, store } from '../redux/store';
import { SET_FILTER, SET_INITIAL_STATE } from '../redux/actions';

function Units() {
  let state = store.getState();
  let { FoodIsActive, WoodIsActive, GoldIsActive } = state.costsActive;

  useLayoutEffect(() => {
    if (FoodIsActive || GoldIsActive || WoodIsActive || state.st === 'B') {
      action(SET_FILTER, { ...state } );
      action(SET_INITIAL_STATE,null);
    }
  }, [action, state.costsActive]);
  return (
    <Fragment>
      <Header title="Units Page" />
      <div className="container">
        <AgeFilter />
        <CostFilter />
        <FiteredTable />
      </div>
    </Fragment>
  );
}

export default Units;
