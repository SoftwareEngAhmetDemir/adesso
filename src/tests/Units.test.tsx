import { shallow } from 'enzyme';
import Units from '../components/Units';
import { render, screen } from '@testing-library/react';

test('should test Units component', () => {
 const wrapper = shallow(<Units  />);
 expect(wrapper).toMatchSnapshot();
});

// test('renders learn react link', () => {
//     render(<Units />);
//     const linkElement = screen.getByText(/Units Page/i);
//     expect(linkElement).toBeInTheDocument();
// });

