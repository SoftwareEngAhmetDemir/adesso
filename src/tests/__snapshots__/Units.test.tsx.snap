// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Units component 1`] = `
<Fragment>
  <Header
    title="Units Page"
  />
  <div
    className="container"
  >
    <AgeFilter />
    <CostFilter />
    <FiteredTable />
  </div>
</Fragment>
`;
